### ISSUE TITLE
Failed to parse kubectl output: com.google.gson.stream.MalformedJsonException

### ISSUE THREAD

**Issue by papanito** [spinnaker/spinnaker#4166]
### Issue Summary:

I get the following error in the `clouddriver` logs

```
Failed to parse kubectl output: com.google.gson.stream.MalformedJsonException
```

### Cloud Provider(s):

On-premise kubernetes

### Environment:

CentOS 7
air-gapped

### Feature Area (if this issue is UI/UX related, please tag `@spinnaker/ui-ux-team`):

clouddriver

### Description:

By chance I have seen this error in the log of the `clouddriver`. It seems however the connection to the k8s cluster is fine.

```
2019-03-21 09:18:45.929  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Describing items in nexus-dev/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:18:45.929  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Caching 0 tagged images in nexus-dev/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:18:45.929  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Caching 0 image ids in nexus-dev/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:18:45.934  INFO 1 --- [ecutionAction-2] c.n.s.c.cache.LoggingInstrumentation     : com.netflix.spinnaker.clouddriver.docker.registry.provider.DockerRegistryProvider:nexus-dev/DockerRegistryImageCachingAgent[1/1] completed in 0.081s
2019-03-21 09:18:45.950  INFO 1 --- [ecutionAction-4] c.n.s.c.cache.LoggingInstrumentation     : kubernetes:dev/KubernetesCoreCachingAgent[1/1] completed in 56.097s
2019-03-21 09:18:46.855  INFO 1 --- [ecutionAction-4] k.v.c.a.KubernetesV2OnDemandCachingAgent : dev/KubernetesCoreCachingAgent[1/1]: agent is starting
2019-03-21 09:18:50.880  WARN 1 --- [ecutionAction-3] s.c.k.v.c.a.KubernetesMetricCachingAgent : dev/KubernetesMetricCachingAgent[1/1]: Metrics for namespace 'sparkpipeline2' in account 'dev' have not been recorded yet.
2019-03-21 09:19:00.138  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Describing items in nexus-private/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:19:00.169  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Caching 2942 tagged images in nexus-private/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:19:00.169  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Caching 2942 image ids in nexus-private/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:19:00.365  INFO 1 --- [ecutionAction-2] c.n.s.c.cache.LoggingInstrumentation     : com.netflix.spinnaker.clouddriver.docker.registry.provider.DockerRegistryProvider:nexus-private/DockerRegistryImageCachingAgent[1/1] completed in 0.511s
2019-03-21 09:19:02.234  INFO 1 --- [0.0-7002-exec-5] c.n.s.c.controllers.SearchController     : Fetching search results for jenkins-slaves-firebird, platform: null, type: [securityGroups], pageSize: 500, pageNumber: 1, filters: [:]
2019-03-21 09:19:02.235  INFO 1 --- [0.0-7002-exec-5] c.n.s.c.k.v.c.KubernetesV2SearchProvider : Querying [securityGroups] for term jenkins-slaves-firebird
2019-03-21 09:19:02.284  INFO 1 --- [0.0-7002-exec-5] c.n.s.c.cache.CatsSearchProvider         : Querying [securityGroups] for term: jenkins-slaves-firebird
2019-03-21 09:19:02.285  INFO 1 --- [0.0-7002-exec-5] c.n.s.c.cache.CatsSearchProvider         : Paginating 0 results; page number: 1, items per page: 500
2019-03-21 09:19:05.017  INFO 1 --- [ecutionAction-3] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesMetricCachingAgent[1/1]: grouping kubernetes_metric has 145 entries and 145 relationships
2019-03-21 09:19:05.028  INFO 1 --- [ecutionAction-3] c.n.s.c.cache.LoggingInstrumentation     : kubernetes:dev/KubernetesMetricCachingAgent[1/1] completed in 20.172s
2019-03-21 09:19:06.857  INFO 1 --- [ecutionAction-3] c.n.s.c.k.v.c.a.KubernetesV2CachingAgent : dev/KubernetesNamespaceCachingAgent[1/1]: agent is starting
2019-03-21 09:19:07.565  INFO 1 --- [ecutionAction-3] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesNamespaceCachingAgent[1/1]: grouping namespace has 38 entries and 0 relationships
2019-03-21 09:19:07.566  INFO 1 --- [ecutionAction-3] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesNamespaceCachingAgent[1/1]: grouping applications has 32 entries and 38 relationships
2019-03-21 09:19:07.573  INFO 1 --- [ecutionAction-3] c.n.s.c.cache.LoggingInstrumentation     : kubernetes:dev/KubernetesNamespaceCachingAgent[1/1] completed in 0.715s
2019-03-21 09:19:08.403  WARN 1 --- [onPool-worker-0] c.n.s.c.k.v.c.a.KubernetesV2CachingAgent : dev/KubernetesCoreCachingAgent[1/1]: Failed to read kind [apiService, clusterRole, clusterRoleBinding, configMap, controllerRevision, customResourceDefinition, cronJob, daemonSet, deployment, event, horizontalpodautoscaler, ingress, job, mutatingWebhookConfiguration, namespace, networkPolicy, persistentVolume, persistentVolumeClaim, pod, podSecurityPolicy, podDisruptionBudget, replicaSet, role, roleBinding, secret, service, serviceAccount, statefulSet, storageClass, validatingWebhookConfiguration] from namespace ingress-nginx: Failed to parse kubectl output: com.google.gson.stream.MalformedJsonException: Unterminated string at line 34037 column 18 path $.items[455]..restuser
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping Policy has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping kubernetes has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping stdio has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping FelixConfiguration has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping kubernetesenv has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping MeshPolicy has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping networkPolicy has 64 entries and 161 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping DestinationRule has 8 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping rule has 6 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping ClusterInformation has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping ServiceEntry has 6 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping Jenkins has 1 entries and 1 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping attributemanifest has 2 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping metric has 6 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping logentry has 2 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping Gateway has 4 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping AppProject has 2 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping prometheus has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping IPPool has 1 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping VirtualService has 17 entries and 0 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping clusters has 9 entries and 64 relationships
2019-03-21 09:19:12.344  INFO 1 --- [ecutionAction-1] s.c.k.v.c.a.KubernetesCacheDataConverter : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: grouping applications has 48 entries and 125 relationships
2019-03-21 09:19:12.555  INFO 1 --- [ecutionAction-1] c.n.s.c.cache.LoggingInstrumentation     : kubernetes:dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1] completed in 59.702s
2019-03-21 09:19:12.853  INFO 1 --- [ecutionAction-1] k.v.c.a.KubernetesV2OnDemandCachingAgent : dev/KubernetesUnregisteredCustomResourceCachingAgent[1/1]: agent is starting
2019-03-21 09:19:14.861  INFO 1 --- [ecutionAction-3] s.c.k.v.c.a.KubernetesMetricCachingAgent : dev/KubernetesMetricCachingAgent[1/1]: agent is starting
2019-03-21 09:19:15.855  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Describing items in nexus-dev/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:19:15.855  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Caching 0 tagged images in nexus-dev/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:19:15.855  INFO 1 --- [ecutionAction-2] .d.r.p.a.DockerRegistryImageCachingAgent : Caching 0 image ids in nexus-dev/DockerRegistryImageCachingAgent[1/1]
2019-03-21 09:19:15.857  INFO 1 --- [ecutionAction-2] c.n.s.c.cache.LoggingInstrumentation     : com.netflix.spinnaker.clouddriver.docker.registry.provider.DockerRegistryProvider:nexus-dev/DockerRegistryImageCachingAgent[1/1] completed in 0.004s
2019-03-21 09:19:17.438  WARN 1 --- [ecutionAction-4] c.n.s.c.cache.LoggingInstrumentation     : kubernetes:dev/KubernetesCoreCachingAgent[1/1] completed with one or more failures
com.netflix.spinnaker.clouddriver.kubernetes.v2.op.job.KubectlJobExecutor$KubectlException: Failed to parse kubectl output: com.google.gson.stream.MalformedJsonException: Unterminated string at line 34037 column 18 path $.items[455]..restuser
	at com.netflix.spinnaker.clouddriver.kubernetes.v2.op.job.KubectlJobExecutor.list(KubectlJobExecutor.java:405) ~[clouddriver-kubernetes.jar:na]
	at com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials.lambda$list$10(KubernetesV2Credentials.java:509) ~[clouddriver-kubernetes.jar:na]
	at com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials.runAndRecordMetrics(KubernetesV2Credentials.java:571) ~[clouddriver-kubernetes.jar:na]
	at com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials.list(KubernetesV2Credentials.java:509) ~[clouddriver-kubernetes.jar:na]
	at com.netflix.spinnaker.clouddriver.kubernetes.v2.caching.agent.KubernetesV2CachingAgent.lambda$loadPrimaryResourceList$0(KubernetesV2CachingAgent.java:94) ~[clouddriver-kubernetes.jar:na]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[na:1.8.0_191]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382) ~[na:1.8.0_191]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481) ~[na:1.8.0_191]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471) ~[na:1.8.0_191]
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:747) ~[na:1.8.0_191]
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:721) ~[na:1.8.0_191]
	at java.util.stream.AbstractTask.compute(AbstractTask.java:316) ~[na:1.8.0_191]
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731) ~[na:1.8.0_191]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[na:1.8.0_191]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) ~[na:1.8.0_191]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) ~[na:1.8.0_191]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) ~[na:1.8.0_191]
Caused by: com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Unterminated string at line 34037 column 18 path $.items[455]..restuser
	at com.google.gson.Gson.fromJson(Gson.java:902) ~[gson-2.7.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:852) ~[gson-2.7.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:801) ~[gson-2.7.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:773) ~[gson-2.7.jar:na]
	at com.netflix.spinnaker.clouddriver.kubernetes.v2.op.job.KubectlJobExecutor.list(KubectlJobExecutor.java:402) ~[clouddriver-kubernetes.jar:na]
	... 16 common frames omitted
Caused by: com.google.gson.stream.MalformedJsonException: Unterminated string at line 34037 column 18 path $.items[455]..restuser
	at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1559) ~[gson-2.7.jar:na]
	at com.google.gson.stream.JsonReader.nextQuotedValue(JsonReader.java:1017) ~[gson-2.7.jar:na]
	at com.google.gson.stream.JsonReader.nextName(JsonReader.java:787) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.ObjectTypeAdapter.read(ObjectTypeAdapter.java:70) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:187) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:145) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129) ~[gson-2.7.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220) ~[gson-2.7.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:887) ~[gson-2.7.jar:na]
	... 20 common frames omitted
```

### Steps to Reproduce:

N/A

### Additional Details:

``` bash
hal --version
1.17.0-20190311152807
```

#### Provider config

```yml
hal config provider kubernetes account get k8sdev
+ Get current deployment
  Success
+ Get k8sdev account
  Success
+ Account k8sdev:
KubernetesAccount(context=dev, cluster=null, user=null, configureImagePullSecrets=true, serviceAccount=null, cacheThreads=1, namespaces=[], omitNamespaces=[], kinds=[], omitKinds=[], customResources=[], cachingPolicies=[], kubeconfigFile=/home/spinnaker/.kube/config, kubeconfigContents=null, kubectlPath=null, kubectlRequestTimeoutSeconds=null, checkPermissionsOnStartup=true, liveManifestCalls=null, oAuthServiceAccount=null, oAuthScopes=[], namingStrategy=null, skin=null, onlySpinnakerManaged=false, debug=null, secretSessionManager=null)
+ K8SDEV

KubernetesAccount(context=dev, cluster=null, user=null,
configureImagePullSecrets=true, serviceAccount=null, cacheThreads=1,
namespaces=[], omitNamespaces=[], kinds=[], omitKinds=[], customResources=[],
cachingPolicies=[], kubeconfigFile=/home/spinnaker/.kube/config,
kubeconfigContents=null, kubectlPath=null, kubectlRequestTimeoutSeconds=null,
checkPermissionsOnStartup=true, liveManifestCalls=null,
oAuthServiceAccount=null, oAuthScopes=[], namingStrategy=null, skin=null,
onlySpinnakerManaged=false, debug=null, secretSessionManager=null)
```

#### kubeconf

Kubeconf is mapped in pod spinnaker-spinnaker-halyard to `~\.kube\conf`

```yml
apiVersion: v1
clusters:
- cluster:
  certificate-authority-data:   <redacted>
  server: https://k8s.intra/k8s/clusters/xxxxx
  name: dev
contexts:
- context:
  cluster: dev
  namespace: spinnaker
  user: token-xxxxx
  name: dev
current-context: dev
kind: Config
preferences: {}
users:
- name: token-xxxxx
  user:
    token: token-xxxxx:<redacted>
```

#### Spinnaker version info

`1.12.5` but with custom bom

```

version: 1.12.5
timestamp: '2019-03-08 23:26:14'
services:
  echo:
	version: 2.3.1-20190214121429
	commit: 5db9d437ca7f2fa374dcada17f77bbbb2965bd67
  clouddriver:
	version: 4.3.5-20190307172446
	commit: f87eb66fd55cd4df7497ef22528a11709745075d
  deck:
	version: 2.7.5-20190308182538
	commit: e9b899d63cb6ea15dc2d6c99a810c8b48886c6a5
  fiat:
	version: 1.3.2-20190128153726
	commit: daf21b24330a5f22866601559aa0f7ac99590274
  front50:
	version: 0.15.2-20190222161456
	commit: 3105e86b8c084ad6ad78507e3a5e5a427f290b99
  gate:
	version: 1.5.2-20190301030607
	commit: b238ab993ab25381ce907260879548ed74a4953f
  igor:
	version: 1.1.1-20190213190226
	commit: 63d06a5c5d55f07443dd60a81035b35cf96238e7
  kayenta:
	version: 0.6.1-20190221030610
	commit: 81d906bf8307143f40fe88f8554baa318de25ef1
  orca:
	version: 2.4.0-20190308182538
	commit: 5e911ff1c29bbc443ce48bcaefbc45f27d389edd
  rosco:
	version: 0.9.0-20190123170846
	commit: 42f81a2501de6d40676d47661579a6106b5c3e8a
  defaultArtifact: {}
  monitoring-third-party:
	version: 0.11.2-20190222030609
	commit: 232c84a8a87cecbc17f157dd180643a8b2e6067a
  monitoring-daemon:
	version: 0.11.2-20190222030609
	commit: 232c84a8a87cecbc17f157dd180643a8b2e6067a
dependencies:
  redis:
	version: 2:2.8.4-2
  consul:
	version: 0.7.5
  vault:
	version: 0.7.0
artifactSources:     
  debianRepository: https://nexus.intra/repository/spinnaker-releases/                
  dockerRegistry: nexus.intra:5000/spinnaker-marketplace           
  googleImageProject: marketplace-spinnaker-release                  
  gitPrefix: https://bitbucket.intra/scm/SPIN/repos
```

**Comment by ezimanyi** [comment]
@papanito : Are you consistently seeing this error for one of your namespaces?  If so, I'd suggest trying to manually run `kubectl` to get the kinds Spinnaker is caching and seeing if the output is indeed valid `json` or what is on the referenced line (`line 34037 column 18` in the error message above) that could be causing the JSON parsing to fail.

If this only happened once or is intermittent, I wonder if it's from the `kubectl` job exiting before fully parsing the JSON.

**Comment by ezimanyi** [comment]
Actually as I think about this some more, it's also possible this is hitting a threading bug in `JobExecutorLocal`.  When getting the [output of a job](https://github.com/spinnaker/clouddriver/blob/f87eb66fd55cd4df7497ef22528a11709745075d/clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/jobs/local/JobExecutorLocal.groovy#L123), we read a `ByteArrayOutputStream` that's currently being written to by another thread, then check the status of the job to see if it's finished.

If either (1) the job finishes between when we store the output and when we check the status, or (2) the thread checking the output doesn't yet see the full update produced by the other thread, then we'll report the job complete with partial output.

If that's the cause of this issue, it would be fixed by spinnaker/clouddriver#3361 which reworked the local job executor code.

**Comment by ezimanyi** [comment]
Closing now as this should be fixed by the above-linked issue.
